app/controllers/hyrax/dashboard/collection_members_controller.rb

```diff
@@ -0,0 +1,65 @@
+module Hyrax
+  module Dashboard
+    ## Shows a list of all collections to the admins
+    class CollectionMembersController < Hyrax::My::CollectionsController
+      before_action :filter_docs_with_read_access!
+
+      include Hyrax::Collections::AcceptsBatches
+
+      def after_update
+        respond_to do |format|
+          format.html { redirect_to success_return_path, notice: 'Collection was successfully updated.' }
+          format.json { render json: @collection, status: :updated, location: dashboard_collection_path(@collection) }
+        end
+      end
+
+      def after_update_error(err_msg)
+        respond_to do |format|
+          format.html { redirect_to err_return_path, alert: err_msg }
+          format.json { render json: @collection.errors, status: :unprocessable_entity }
+        end
+      end
+
+      def update_members
+        err_msg = validate
+        after_update_error(err_msg) if err_msg.present?
+        return if err_msg.present?
+
+        collection.add_member_objects batch_ids
+        after_update
+      end
+
+      private
+
+        def validate
```

Comment:
My non-blocking expectation is that this behavior would be captured in a form object. However, I accept that that is not the case and am okay with proceeding as written.